src/LibraryManager/LibrariesValidator.cs

```diff
@@ -24,141 +28,346 @@ internal class LibrariesValidator
         /// <param name="defaultProvider"></param>
         public LibrariesValidator(IDependencies dependencies, string defaultDestination, string defaultProvider)
         {
-            _dependencies = dependencies ?? throw new ArgumentNullException(nameof(dependencies));
+            _dependencies = dependencies;
             _defaultDestination = defaultDestination;
             _defaultProvider = defaultProvider;
         }
-
-        private string _defaultDestination;
-        private string _defaultProvider;
-        private IDependencies _dependencies;
-
+        
         /// <summary>
-        /// Returns a collection of ILibraryOperationResult that represents the status for validation of each 
-        ///  library 
+        /// Validates all properties of the library including detection of invalid files
         /// </summary>
-        /// <param name="libraries"></param>
+        /// <param name="library"></param>
         /// <param name="cancellationToken"></param>
         /// <returns></returns>
-        public async Task<IEnumerable<ILibraryOperationResult>> GetLibrariesErrorsAsync(
-            IEnumerable<ILibraryInstallationState> libraries,
-            CancellationToken cancellationToken)
+        public async Task<ILibraryOperationResult> ValidateLibraryAsync(ILibraryInstallationState library, CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested();
 
-            IEnumerable<ILibraryOperationResult> validateLibraries = ValidatePropertiesAsync(libraries, cancellationToken);
+            var list = new List<IError>();
 
-            if (!validateLibraries.All(t => t.Success))
+            // Validates provider
+            IError error;
+            IProvider provider = GetProvider(library, out error);
+            if (provider == null)
             {
-                return validateLibraries;
+                list.Add(error);
```

Comment:
>error [](start = 25, length = 5)

Should this (or the if condition) check `error`?

Reply:
When provider is null, error is not. Not sure what you meant

---
In reply to: [195515530](https://github.com/aspnet/LibraryManager/pull/164#discussion_r195515530) [](ancestors = 195515530)